### ISSUE TITLE
Improve Markdown support - multiline setext headings are wrongly parsed

### ISSUE THREAD

**Issue by igormironchik** [doxygen/doxygen#10894]
**Describe the bug**

A following heading wrongly parses

```md
I need to know positions in the `Markdown` file of blocks/elements. How
can I achieve this?
---

```

**Expected behavior**

These three lines of Markdown should be one heading.

**Version**

```
1.12.0 (3d494140a674a8344c165a559d841166bbf3b80f*)
```

```
No LSB modules are available.
Distributor ID: Neon
Description:    KDE neon 6.0
Release:        22.04
Codename:       jammy
```

I've built `Doxygen` from sources.

**Comment by igormironchik** [comment]
And corresponding GitHub heading's ID should be like in a following link

```md
  * [I need to know positions in the `Markdown` file of blocks/elements. How can I achieve this?](#i-need-to-know-positions-in-the-markdown-file-of-blockselements-howcan-i-achieve-this)
```

Be careful with handling new line in ID, look at `-howcan-`

**Comment by igormironchik** [comment]
https://spec.commonmark.org/0.31.2/#setext-headings

**Comment by albert-github** [comment]
To be sure we are talking about the same issue always:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

**Comment by igormironchik** [comment]
[Doxygen.zip](https://github.com/doxygen/doxygen/files/15436816/Doxygen.zip)

**Comment by doxygen** [comment]
@albert-github I tend to agree with you that a multiple line heading is probably more often a mistake, than something a user would want, even though the common mark spec supports this. I can imagine existing comments with some `----`  separator would look rather horrible if the whole paragraph is suddenly turned into a section.

**Comment by igormironchik** [comment]
![Screenshot_20240528_093959](https://github.com/doxygen/doxygen/assets/11455533/6892c452-783d-4671-b22b-392f53eb97af)

Are these the long headings?

**Comment by igormironchik** [comment]
![Screenshot_20240528_095613](https://github.com/doxygen/doxygen/assets/11455533/873db44e-642d-4cb7-90ba-e49fd6222f71)

So why should I be restricted to one line?

**Comment by albert-github** [comment]
I think it is not only about the length but most of the time also about the, bad, formulation of the section title.

The section title you mention:
> I need to know positions in the `Markdown` file of blocks / elements,. How can I achieve this?

Is a long and badly formulated heading already indicated due to the fact that there are 2 sentences.
A better formulation would be
> How to obtain the position of blocks / elements in a `Markdown` file?


Sometimes it will be quite hard to shorten / get a better formulation though.

**Comment by igormironchik** [comment]
Guys, I figured out that this is a regression. In `Doxygen` version `1.9.8` this thing works properly.

![1 9 8](https://github.com/doxygen/doxygen/assets/11455533/82cf7146-11d4-4630-86aa-bf22b8c96881)

And in 1.12.0

![1 12 0](https://github.com/doxygen/doxygen/assets/11455533/053f21e9-3f2e-477b-a00e-7130da3f8e6b)

And why it's bad to have such a heading? Look at it in my editor.

![Screenshot_20240529_081331](https://github.com/doxygen/doxygen/assets/11455533/df5188fe-6e57-422d-a473-5ea189ded728)

**Comment by albert-github** [comment]
From images it is hard / not possible to get the proper file to see what causes the problem.
So:
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file!) that allows us to reproduce the problem? Please don't add external links as they might not be persistent (also references to GitHub repositories are considered non persistent).

Furthermore with the original example given in https://github.com/doxygen/doxygen/issues/10894#issuecomment-2129853735 I get with the 1.9.8 version:
![image](https://github.com/doxygen/doxygen/assets/5223533/23a1e106-d9bc-4d15-a2cb-4cfdd51d8c0e)

so this looks regarding the issue not correct either..

**Comment by igormironchik** [comment]
> so this looks regarding the issue not correct either..

Hm, maybe, maybe I changed `HTML` by hands a long time ago, and this issue is very old.
